client/components/AddGameModal.tsx

```diff
@@ -228,7 +228,10 @@ const SetupGameForm: React.FC<{ onClose: () => void }> = ({ onClose }) => {
                     colorScheme="teal"
                     isLoading={loading}
                     disabled={loading || game.syncing}
-                    onClick={() => onAdd({ name, continueSearching })}
+                    onClick={() => {
+                        onAdd({ name, continueSearching });
+                        onSave();
+                    }}
```

Comment:
A better approach would be to replace the `onSuccess: onClose` in line 132 with `onSucess: onSave` so that we only call this method if the `onAdd` really worked. With the current implementation here we do not wait for the promise of `onAdd` to resolve